src/Hazelcast.Net/Predicates/Predicate.cs

```diff
@@ -12,119 +12,86 @@
-        }
-
-        public static AndPredicate And(params IPredicate[] predicates)
-        {
-            return new AndPredicate(predicates);
-        }
-
-        public static FalsePredicate False()
-        {
-            return new FalsePredicate();
-        }
-
-        public static BetweenPredicate IsBetween(string attributeName, object from, object to)
-        {
-            return new BetweenPredicate(attributeName, from, to);
-        }
-
-        public static EqualPredicate IsEqual(string attributeName, object value)
-        {
-            return new EqualPredicate(attributeName, value);
-        }
-
-        public static GreaterLessPredicate IsGreaterThan(string attributeName, object value)
-        {
-            return new GreaterLessPredicate(attributeName, value, false, false);
-        }
-
-        public static GreaterLessPredicate IsGreaterThanOrEqual(string attributeName, object value)
-        {
-            return new GreaterLessPredicate(attributeName, value, true, false);
-        }
-
-        public static CaseInsensitiveLikePredicate IsILike(string attributeName, string expression)
-        {
-            return new CaseInsensitiveLikePredicate(attributeName, expression);
-        }
-
-        public static InPredicate IsIn(string attributeName, params object[] values)
-        {
-            return new InPredicate(attributeName, values);
-        }
-
-        public static GreaterLessPredicate IsLessThan(string attributeName, object value)
-        {
-            return new GreaterLessPredicate(attributeName, value, false, true);
-        }
-
-        public static GreaterLessPredicate IsLessThanOrEqual(string attributeName, object value)
-        {
-            return new GreaterLessPredicate(attributeName, value, true, true);
-        }
-
-        public static LikePredicate IsLike(string attributeName, string expression)
-        {
-            return new LikePredicate(attributeName, expression);
-        }
-
-        public static NotEqualPredicate IsNotEqual(string attributeName, object value)
-        {
-            return new NotEqualPredicate(attributeName, value);
-        }
-
-        public static RegexPredicate MatchesRegex(string attributeName, string regex)
-        {
-            return new RegexPredicate(attributeName, regex);
-        }
-
-        public static NotPredicate Not(IPredicate predicate)
-        {
-            return new NotPredicate(predicate);
-        }
-
-        public static OrPredicate Or(params IPredicate[] predicates)
-        {
-            return new OrPredicate(predicates);
-        }
-
-        public static SqlPredicate Sql(string sql)
-        {
-            return new SqlPredicate(sql);
-        }
-
-        public static TruePredicate True()
-        {
-            return new TruePredicate();
-        }
+            => new PredicateProperty(ThisConst);
+
+        public static IPredicate InstanceOf(string fullJavaClassName)
```

Comment:
I am not sure about that, but if we have `IsXX` methods below, should we also rename this to `IsInstanceOf` for consistency? 

By the way, I prefer shorter names like this. For example, I would also change  `IsGreaterThanOrEqual` to `GreaterOrEqual`, but I guess explicit names are a more common thing in C#. So, the current state might be more appropriate for C#

Reply:
there's a later PR I'm working on that normalizes these names, so I won't change this here